protocol/bc/legacy/map.go

```diff
@@ -143,17 +145,10 @@ func mapTx(tx *TxData) (headerID bc.Hash, hdr *bc.TxHeader, entryMap map[bc.Hash
 
 			var (
 				anchorID    bc.Hash
-				setAnchored func(*bc.Hash)
+				setAnchored = func(*bc.Hash) {}
```

Comment:
It's a pretty common way to avoid nil checks in Go. Ex: [database/sql example](https://github.com/golang/go/blob/release-branch.go1.8/src/database/sql/sql.go#L1728), [crypto/tls example](https://github.com/golang/go/blob/release-branch.go1.8/src/crypto/tls/tls.go#L194)